packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -0,0 +1,222 @@
+    sortIndicator,
+    hasError,
+    isResizable,
+    isSelectable,
+    tableHeight,
+    width,
+    minimumWidth,
+    maximumWidth,
+    headerHeight,
+    intl,
+    columnIndex,
+    onColumnSelect,
+    onResizeMouseDown,
+    rowIndex,
+  } = props;
+
+  const columnContext = useContext(ColumnContext);
+  const theme = useContext(ThemeContext);
+  const gridContext = useContext(GridContext);
+  const columnHeaderCell = useRef();
+
+  const onResizeHandleMouseDown = useCallback((event) => {
+    if (onResizeMouseDown) {
+      onResizeMouseDown(event, columnIndex, columnHeaderCell.current.offsetWidth);
+    }
+  }, [columnIndex, onResizeMouseDown]);
+
+  // Handle column header selection via the mouse click.
+  const handleMouseDown = (event) => {
+    onColumnSelect(id, { row: rowIndex, col: columnIndex });
+    event.stopPropagation();
+  };
+
+  const handleKeyDown = (event) => {
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        onColumnSelect(id, { row: rowIndex, col: columnIndex });
+        event.stopPropagation();
+        event.preventDefault(); // prevent the default scrolling
+        break;
+      default:
+    }
+  };
+
+  const errorIcon = hasError && <IconError a11yLabel={intl.formatMessage({ id: 'Terra.table.columnError' })} className={cx('error-icon')} />;
+
+  let sortIndicatorIcon;
+  if (sortIndicator === SortIndicators.ASCENDING) {
+    sortIndicatorIcon = <IconUp />;
+  } else if (sortIndicator === SortIndicators.DESCENDING) {
+    sortIndicatorIcon = <IconDown />;
+  }
+
+  const cellLeftEdge = (columnIndex < columnContext.pinnedColumnOffsets.length) ? columnContext.pinnedColumnOffsets[columnIndex] : null;
+  const dividerLeftEdge = width - 1;
+
+  const pinnedColumnsDivider = columnIndex === columnContext.pinnedColumnOffsets.length - 1
+    ? (
+      <div
+        className={cx('pinned-columns-divider')}
+        style={{ height: tableHeight, left: dividerLeftEdge }} // eslint-disable-line react/forbid-dom-props
+      />
+    )
+    : null;
+
+  return (
+  /* eslint-disable react/forbid-dom-props */
+    <th
+      ref={columnHeaderCell}
+      key={id}
+      className={cx('column-header', theme.className, { selectable: isSelectable, pinned: columnIndex < columnContext.pinnedColumnOffsets.length })}
+      role="columnheader"
+      scope="col"
+      tabIndex={gridContext.role === 'grid' ? -1 : undefined}
+      aria-sort={sortIndicator}
+      onMouseDown={(isSelectable && onColumnSelect) ? handleMouseDown : undefined}
+      onKeyDown={isSelectable && onColumnSelect ? handleKeyDown : undefined}
+      // eslint-disable-next-line react/forbid-dom-props
+      style={{ width: `${width}px`, height: headerHeight, left: cellLeftEdge }}
+    >
+      {/* eslint-disable jsx-a11y/no-noninteractive-tabindex */}
+      <div className={cx('header-container')} role={displayName && 'button'} tabIndex={isSelectable ? 0 : undefined}>
```

Comment:
You have the tabIndex at two separate levels.

Reply:
Stripped this out, since we don't want to announce buttons in VO/Jaws anyways.